CondFormats/JetMETObjects/interface/JetResolutionObject.h

```diff
@@ -0,0 +1,335 @@
+                    std::vector<Binning> m_variables COND_TRANSIENT;
+
+                    COND_SERIALIZABLE;
+            };
+
+            class Record {
+                public:
+                    Record() {
+                        // Empty
+                    }
+
+                    Record(const std::string& record, const Definition& def);
+
+                    const std::vector<Range>& getBinsRange() const {
+                        return m_bins_range;
+                    }
+
+                    const std::vector<Range>& getVariablesRange() const {
+                        return m_variables_range;
+                    }
+
+                    const std::vector<float>& getParametersValues() const {
+                        return m_parameters_values;
+                    }
+
+                    size_t nVariables() const {
+                        return m_variables_range.size();
+                    }
+
+                    size_t nParameters() const {
+                        return m_parameters_values.size();    
+                    }
+
+                private:
+                    std::vector<Range> m_bins_range;
+                    std::vector<Range> m_variables_range;
+                    std::vector<float> m_parameters_values;
+
+                    COND_SERIALIZABLE;
+            };
+
+        public:
+            JetResolutionObject(const std::string& filename);
+            JetResolutionObject(const JetResolutionObject& filename);
+            JetResolutionObject();
+
+            void dump() const;
+            void saveToFile(const std::string& file) const;
+
+            const Record* getRecord(const JetParameters& bins) const;
+            float evaluateFormula(const Record& record, const JetParameters& variables) const;
+
+            const std::vector<Record>& getRecords() const {
+                return m_records;
+            }
+
+            const Definition& getDefinition() const {
+                return m_definition;
+            }
+
+        private:
+            Definition m_definition;
+            std::vector<Record> m_records;
+
+            bool m_valid = false;
+
+            COND_SERIALIZABLE;
+    };
+
+
+    class JetResolution {
+        public:
+            JetResolution(const std::string& filename);
+            JetResolution(const JetResolutionObject& object);
+            JetResolution() {
+                // Empty
+            }
+
+#ifndef STANDALONE
+            static const JetResolution get(const edm::EventSetup&, const std::string&);
+#endif
+
+            float getResolution(const JetParameters& parameters) const;
+
+            void dump() const {
+                m_object->dump();
+            }
+
+            // Advanced usage
+            const JetResolutionObject* getResolutionObject() const {
+                return m_object.get();
+            }
+
+        private:
+            std::shared_ptr<JetResolutionObject> m_object;
+    };
+
+    class JetResolutionScaleFactor {
```

Comment:
Sorry for the delay, I was in holidays.

Do you want me to move these classes into another files?